File: CodeSlice.UnitTesting/CodeSlice.UnitTesting.Model/DomainEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace CodeSlice.UnitTesting.Model
{
    /// <summary>
    /// Base class for domain entities responsible for holding auditing and 
    /// persistence related properties
    /// </summary>
    public abstract class DomainEntity
    {

        /// <summary>
        /// Default Constructor that sets the created date
        /// </summary>
        public DomainEntity()
        {
            CreatedDate = DateTime.Now;
        }

        [Key]
        public int Id { get; set; }

        [Required]
        public string CreatedBy { get; set; }

        [Required]
        public DateTime CreatedDate { get; private set; }
    }
}

File: CodeSlice.UnitTesting/CodeSlice.UnitTesting.Web/src/DomainEntity.js
Ext.define('DomainEntity', {
    config: {
        id:         null,
        createdBy:   null,
        createdDate: new Date()
    },

    constructor: function(config){
        this.initConfig(config);
        return this;
    },

    validate: function(){
        var errors = [];

        if(this.id === null){
            errors.push({ field: 'id', message: "ID cannot be null" });
        }

        if(this.createdBy === null || this.createdBy === ''){
            errors.push({ field: 'createdBy', message: "Created By cannot be null" });
        }

        if(this.createdDate === null){
            errors.push({ field: 'createdDate', message: "Created Date cannot be null" });
        }

        return errors;
    }
});